Sources/MongoSwift/ConnectionPool.swift

```diff
@@ -45,6 +45,15 @@ internal class ConnectionPool {
             throw InternalError(message: "Could not configure error handling on client pool")
         }
 
+        if let maxPoolSize = options?.maxPoolSize {
+            guard maxPoolSize > 0 && maxPoolSize <= UInt32.max else {
+                throw InvalidArgumentError(
```

Comment:
We could just specify the option in `ClientOptions` as a `UInt32` to avoid this check.

Reply:
I considered this. however, I think the community seems to have converged on still using `Int` in cases like this. Consider that, for example, `NIOThreadPool.init` and `MultiThreadedEventLoopGroup` initializers both take in `Int`s even though those values definitely need to be positive.
We do currently have the additional restriction of max pool size being capped at `UInt32.max`, but that's a constraint introduced by libmongoc. It doesn't seem like we would bother to cap the max allowed pool size at that value in a pure Swift implementation.

Also from the [Swift language guide](https://docs.swift.org/swift-book/LanguageGuide/TheBasics.html) (see UInt section):
> Use `UInt` only when you specifically need an unsigned integer type with the same size as the platform’s native word size. If this isn’t the case, `Int` is preferred, even when the values to be stored are known to be nonnegative. A consistent use of `Int` for integer values aids code interoperability, avoids the need to convert between different number types, and matches integer type inference, as described in Type Safety and Type Inference.

That said, if everyone is on board with adopting this guidance we probably need to do an audit of our API to correct existing violations.

Reply:
Oh cool, I didn't know about that section of the guide, good find. And yeah, we probably should adopt that guidance, since others likely are (e.g. NIO is).